### PR TITLE
ITS: TrackExtensionStudy

### PR DESCRIPTION
**Pull request by f3sch**
The majority oft the changes extends the track extension study code. The rest makes minor modifications which should be safe. Also this fixes one bug for the track following where the vector iterator gets invalidate when the vector is reallocated.

### CODE REVIEW

**Comment by f3sch** [comment]
Undrafting to check CI.

File: Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# This software is distributed under the terms of the GNU General Public
+# License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+#
+# In applying this license CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+o2_add_test_root_macro(
+  PostTrackExtension.C
+  PUBLIC_LINK_LIBRARIES ROOT::Hist ROOT::RIO ROOT::Core ROOT::Gpad
```
Comment (mconcas) [inline Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt]: Maybe you need:
```suggestion
  PUBLIC_LINK_LIBRARIES ROOT::Hist ROOT::RIO ROOT::Core ROOT::Gpad fmt::fmt
```

Comment (f3sch) [reply Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt]: Curiously, it still complains about it.
```
cling::DynamicLibraryManager::loadLibrary(): dlopen(libfmt.10.1.0.dylib, 0x0009): tried: 'libfmt.10.1.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibfmt.10.1.0.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/../lib/libfmt.10.1.0.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/bin/../lib/libfmt.10.1.0.dylib' (no such file), '/usr/lib/libfmt.10.1.0.dylib' (no such file, not in dyld cache), 'libfmt.10.1.0.dylib' (no such file)
cling::DynamicLibraryManager::loadLibrary(): dlopen(/Volumes/build/alice-ci-workdir/o2/sw/BUILD/e89aa9f3d18dea00fd76f985216d2b4e905e7c6c/O2/compiled_macros/Volumes/build/alice-ci-workdir/o2/sw/SOURCES/O2/13568/0/Detectors/ITSMFT/ITS/postprocessing/studies/macros/PostTrackExtension_C.so, 0x0009): Library not loaded: @rpath/libfmt.10.dylib
  Referenced from: <7E867859-82DA-3B84-9471-92E6AA6D43A1> /Volumes/build/alice-ci-workdir/o2/sw/BUILD/e89aa9f3d18dea00fd76f985216d2b4e905e7c6c/O2/compiled_macros/Volumes/build/alice-ci-workdir/o2/sw/SOURCES/O2/13568/0/Detectors/ITSMFT/ITS/postprocessing/studies/macros/PostTrackExtension_C.so
  Reason: tried: '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/../lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/bin/../lib/libfmt.10.dylib' (no such file)
ld: Undefined symbols:
```

Comment (mconcas) [reply Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt]: > Curiously, it still complains about it.
> 
> ```
> cling::DynamicLibraryManager::loadLibrary(): dlopen(libfmt.10.1.0.dylib, 0x0009): tried: 'libfmt.10.1.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibfmt.10.1.0.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/../lib/libfmt.10.1.0.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/bin/../lib/libfmt.10.1.0.dylib' (no such file), '/usr/lib/libfmt.10.1.0.dylib' (no such file, not in dyld cache), 'libfmt.10.1.0.dylib' (no such file)
> cling::DynamicLibraryManager::loadLibrary(): dlopen(/Volumes/build/alice-ci-workdir/o2/sw/BUILD/e89aa9f3d18dea00fd76f985216d2b4e905e7c6c/O2/compiled_macros/Volumes/build/alice-ci-workdir/o2/sw/SOURCES/O2/13568/0/Detectors/ITSMFT/ITS/postprocessing/studies/macros/PostTrackExtension_C.so, 0x0009): Library not loaded: @rpath/libfmt.10.dylib
>   Referenced from: <7E867859-82DA-3B84-9471-92E6AA6D43A1> /Volumes/build/alice-ci-workdir/o2/sw/BUILD/e89aa9f3d18dea00fd76f985216d2b4e905e7c6c/O2/compiled_macros/Volumes/build/alice-ci-workdir/o2/sw/SOURCES/O2/13568/0/Detectors/ITSMFT/ITS/postprocessing/studies/macros/PostTrackExtension_C.so
>   Reason: tried: '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/lib/../lib/libfmt.10.dylib' (no such file), '/Volumes/build/alice-ci-workdir/o2/sw/osx_arm64/ROOT/v6-32-06-alice1-local3/bin/../lib/libfmt.10.dylib' (no such file)
> ld: Undefined symbols:
> ```

This is the exact error you had before, right? So my suggestion was pointless 👎🏼

Comment (f3sch) [reply Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt]: Well correct to link it nonetheless but yes it did not fix it and stayed the same. But I kind of am out of ideas what else to add, must be something insanely obvious

Comment (f3sch) [reply Detectors/ITSMFT/ITS/postprocessing/studies/macros/CMakeLists.txt]: @ktf Sorry to bother you but do you have an idea what I am missing in the linking?

**Comment by f3sch** [comment]
@mconcas failures are unrelated now, would you be fine to merge this?

**Pull request merged**